Source/WebCore/html/HTMLMetaElement.cpp

```diff
@@ -106,11 +106,19 @@ void HTMLMetaElement::attributeChanged(const QualifiedName& name, const AtomStri
         if (isInDocumentTree()) {
             if (equalLettersIgnoringASCIICase(oldValue, "theme-color"_s) && !equalLettersIgnoringASCIICase(newValue, "theme-color"_s))
                 document().metaElementThemeColorChanged(*this);
+#if ENABLE(DARK_MODE_CSS)
+            else if (equalLettersIgnoringASCIICase(oldValue, "color-scheme"_s) || equalLettersIgnoringASCIICase(oldValue, "supported-color-schemes"_s))
```

Comment:
It might be time to factor this into a method, since it already appears twice in this file.

Reply:
> It might be time to factor this into a method, since it already appears twice in this file.

Thanks — the update I’ve pushed includes factoring it into a new function (which is a member function because — to deal with preventing the code from getting called twice — also sets a private data member).